ui-personalinfo/src/main/kotlin/edu/stanford/spezi/ui/personalinfo/UserProfile.kt

```diff
@@ -27,24 +27,46 @@ import androidx.compose.ui.unit.sp
 import edu.stanford.spezi.core.logging.SpeziLogger
 import edu.stanford.spezi.ui.Colors
 import edu.stanford.spezi.ui.ImageResource
-import edu.stanford.spezi.ui.ImageResourceComposable
 import edu.stanford.spezi.ui.SpeziTheme
 import edu.stanford.spezi.ui.StringResource
 import edu.stanford.spezi.ui.ThemePreviews
 import edu.stanford.spezi.ui.lighten
 import kotlin.math.min
 
+interface UserProfileImageLoader {
+    suspend fun loadImage(): ImageResource?
+}
+
 @Composable
-fun UserProfileComposable(
+fun UserProfile(
     name: PersonNameComponents,
     modifier: Modifier = Modifier,
     imageLoader: suspend () -> ImageResource? = { null },
+) {
+    UserProfile(
+        name = name,
+        modifier = modifier,
+        imageLoader = remember(imageLoader) {
+            object : UserProfileImageLoader {
+                override suspend fun loadImage(): ImageResource? {
+                    return imageLoader()
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun UserProfile(
+    name: PersonNameComponents,
+    modifier: Modifier = Modifier,
+    imageLoader: UserProfileImageLoader?,
 ) {
     var size by remember { mutableStateOf(IntSize.Zero) }
     var loadedImage by remember(imageLoader) { mutableStateOf<ImageResource?>(null) }
 
-    LaunchedEffect(Unit) {
-        loadedImage = runCatching { imageLoader() }
+    LaunchedEffect(imageLoader) {
```

Comment:
Again, this setup would not be needed if we use coil, as it will make sure to launch coroutines and take care of caching internally

Reply:
See above